scipy/fft/_fftlog.py

```diff
@@ -110,6 +110,68 @@ def fht(a, dln, mu, offset=0.0, bias=0.0):
     .. [1] Talman J. D., 1978, J. Comp. Phys., 29, 35
     .. [2] Hamilton A. J. S., 2000, MNRAS, 312, 257 (astro-ph/9905191)
 
+    Examples
+    --------
+
+    This example is the adapted version of ``fftlogtest.f`` which is provided
+    in [2]_. It evaluates the integral
+
+    .. math::
+
+        \int^\infty_0 r^{\mu+1} \exp(-r^2/2) J_\mu(k, r) k dr
+        = k^{\mu+1} \exp(-k^2/2) .
+
+    >>> from scipy import fft
+    >>> import matplotlib.pyplot as plt
+
+    Parameters for the transform.
+
+    >>> mu = 0.0                    # Order mu of Bessel function
+    >>> r = np.logspace(-4, 4, 64)  # Input evaluation points
+    >>> dln = np.log(r[1]/r[0])     # Step size
+    >>> offset = fft.fhtoffset(dln, mu=mu)  # Optimal offset
+    >>> k = np.exp(offset)/r[::-1]  # Output evaluation points
+
+    Define the theoretical function (same for input and output).
+
+    >>> def theoretical(x, mu):
+    >>>     """Theoretical input/output function: x^(mu+1) exp(-x^2/2)."""
```

Comment:
I think for block continuations like this in the docstring examples you'll want to use `...` instead of `>>>` on the lines after the first line to avoid errors like the one currently shown in CI:

```
File "build/testenv/lib/python3.9/site-packages/scipy/fft/_fftlog_multimethods.py", line ?, in fht
Failed example:
    def theoretical(x, mu):
Exception raised:
    Traceback (most recent call last):
      File "/opt/hostedtoolcache/Python/3.9.13/x64/lib/python3.9/doctest.py", line 1334, in __run
        exec(compile(example.source, filename, "single",
      File "<doctest fht[7]>", line 1
        def theoretical(x, mu):
                               ^
    IndentationError: expected an indented block

```